### ISSUE TITLE
Provide a mechanism wherein a user can plugin their own custom wait mechanism

### ISSUE THREAD

**Issue by krmahadevan** [RationaleEmotions/SimpleSe#13]
SimpleSe has a very limited subset of wait mechanisms. But selenium has a wide variety of waits defined.

So a user should be able to plugin their own wait mechanism into the codebase and simplese should be able to work with the newly plugged in wait mechanism.

**Comment by bharathc02** [comment]
would this solution be clean? The value of until can be the key to custom wait mechanism. 
A map with _key_ as these names and values as _ExpectedCondition_ can be registered. 
```
"wait": {
        "until": "visible",
        "for": 45
      }
```
If there is no corresponding mapping, it can default to _visible_ condition.
Additionally, there can be a condition called _disable_ which can remove wait condition and can also be accessed via _WebPage_ class.

Let me know your thoughts on this and I can give a PR.

**Comment by krmahadevan** [comment]
@bharathc02 - Yes it is pretty much what I had in mind. 

> Additionally, there can be a condition called disable which can remove wait condition and can also be accessed via WebPage class.

Disabling waits is already taken care of, when I fixed #12 I also would like to ensure that `Waits` are not available for access via any means, because a user is not going to be doing anything with it.

I would expect the wire in mechanism to be  [SPI](https://www.baeldung.com/java-spi) driven and SimpleSe maintains the registry and applies the appropriate wait logic. 

If you would like to raise a PR, please go ahead.

**Comment by krmahadevan** [comment]
Fixed via #14
